### DISCUSSION TITLE
Using Tailwind with Deno

### DISCUSSION THREAD

**Discussion by nico-bachner** [tailwindlabs/tailwindcss#3227] [General]
I hope this is not a duplicate, but I have not been able to find any documentation on how to use Tailwind with Deno.
I'm trying both out for the first time and am not sure where to start.

**Comment by aadamsx** [comment]
Is Tailwind compatible with Deno + React?

**Reply by ghost** [reply]
It seems PostCSS should work in Deno, but Tailwind [fails](https://github.com/postui/esm.sh/issues/10).
But I guess you can still use Tailwind with Node to generate the CSS for the Deno project? (like it's done in Rails and others)

**Reply by aadamsx** [reply]
Not a long term solution. And to me not an option.

**Reply by ghost** [reply]
I agree. I was also looking for a better solution, but I noticed you already created an [issue](https://github.com/tailwindlabs/tailwindcss/issues/3832). thanks.

**Reply by lucidNTR** [reply]
the problem is that deno and node have completely different security guarantees and running a node cli tool in your local build would require adding docker or similar sandboxing.

i think the way forward would be to migrate to es modules and add a test run that checks the node apis that the cli uses are compatible with the deno node compatibility layer. this way the work load on tailwind is managable and also will future proof the setup...
